src/main/java/org/kohsuke/stapler/idea/JellyCompletionContributor.java

```diff
@@ -1,146 +1,247 @@
+            JellyElement component, Project project, Editor editor, String prefix) {
+        TemplateManager templateManager = TemplateManager.getInstance(project);
+        StringBuilder templateText = new StringBuilder();
+
+        List<String> requiredAttributes = getRequiredAttributes(component);
+        for (String attributeName : requiredAttributes) {
+            templateText
+                    .append(" ")
+                    .append(attributeName)
+                    .append("=\"$")
+                    .append(attributeName)
+                    .append("$\"");
+        }
+
+        if (component.contentType() == XmlElementDescriptor.CONTENT_TYPE_ANY) {
+            templateText
+                    .append(">\n  $content$\n</")
+                    .append(prefix)
+                    .append(":")
+                    .append(component.name())
+                    .append(">");
+        } else {
+            templateText.append(" />");
+        }
+
+        Template template = templateManager.createTemplate("myTemplate", "Jelly", templateText.toString());
+
+        // Add each attribute as a placeholder variable for tabbing
+        for (String attributeName : requiredAttributes) {
+            template.addVariable(attributeName, "", attributeName.toLowerCase(), true);
+        }
+        if (component.contentType() == XmlElementDescriptor.CONTENT_TYPE_ANY) {
+            template.addVariable("content", "", "content", true);
+        }
+
+        templateManager.startTemplate(editor, template);
+    }
+
+    private static List<String> getRequiredAttributes(JellyElement component) {
+        List<String> requiredAttributes = new ArrayList<>();
+        for (XmlAttributeDescriptor attributesDescriptor : component.attributesDescriptors()) {
+            if (attributesDescriptor.isRequired()) {
+                requiredAttributes.add(attributesDescriptor.getName());
+            }
+        }
+        return requiredAttributes;
+    }
+
+    private static String getPrefix(String tagName) {
+        if (tagName == null || !tagName.contains(":")) {
+            return null;
+        }
+
+        // Remove the opening '<' and check if there is a colon in the string
+        tagName = tagName.replaceAll("^<", "");
+        int colonIndex = tagName.indexOf(':');
+
+        // If there's no colon or the colon is at the start, return null
+        if (colonIndex <= 0) {
+            return null;
+        }
+
+        // Return the prefix, which is the substring before the first ':'
+        return tagName.substring(0, colonIndex);
+    }
+
+    /**
+     * Creates a merged map of namespaces with their corresponding URIs, combining core namespaces, project-specific
+     * namespaces, and namespaces declared in the current file, in that order of precedence.
+     */
+    private static Map<String, String> createMergedNamespaceMap(XmlTag tag) {
+        Map<String, String> projectNamespaces = collectProjectNamespaces(tag.getProject());
```

Comment:
I wonder if the value should be a set in some way.

<img width="319" alt="image" src="https://github.com/user-attachments/assets/04fa9af4-4832-4cd5-af44-93e4fa42ed86">

In core everything gets repeated because some files for whatever reason use a different prefix for the namespace.

Alternatively could be just fixed by a PR to core to be consistent...